src/frontend/src/flows/register.ts

```diff
@@ -59,18 +60,20 @@ const init = (): Promise<LoginResult | null> =>
       try {
         const pendingIdentity = WebAuthnIdentity.create({
           publicKey: creationOptions(),
-        }).catch((error) => {
-          resolve(apiResultToLoginResult({ kind: "authFail", error }));
-          // We can never get here, but TS doesn't understand that
-          return 0 as unknown as WebAuthnIdentity;
         });
         await tick();
         // Kick-start both the captcha creation and the identity
-        const captcha = makeCaptcha();
-        const identity = await pendingIdentity;
-        await captcha;
-        const result = await confirmRegister(captcha, identity, alias);
-        resolve(result);
+        Promise.all([makeCaptcha(), pendingIdentity])
+          .catch((error) => {
+            resolve(apiResultToLoginResult({ kind: "authFail", error }));
+            // We can never get here, but TS doesn't understand that
+            return 0 as unknown as [Challenge, WebAuthnIdentity];
+          })
+          .then(([captcha, identity]) => {
+            confirmRegister(Promise.resolve(captcha), identity, alias).then(
```

Comment:
The captcha has already been resolved by virtue of being an output of promise.all.  If all were successful, it has been resolved, if one failed, the others will have been aborted.

Reply:
I know, but I need it to typecheck :)